src/game/Board.java

```diff
@@ -0,0 +1,376 @@
+package game;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import com.nedap.go.gui.GoGUIIntegrator;
+
+import exceptions.ExitProgram;
+
+/**
+ * The board on which a game of Go is played.
+ */
+public class Board {
+
+	/**
+	 * The size of the board.
+	 */
+	private int boardSize;
+
+	/**
+	 * The intersections of this board. Can either be black/white stones or
+	 * unoccupied (set to public for testing).
+	 */
+	public Intersec[] intersecs;
+
+	/**
+	 * The GUI of this board.
+	 */
+	private GoGUIIntegrator g;
+
+	/**
+	 * Indicates whether a first stone has been placed on the board.
+	 */
+	private boolean firstStone;
+
+	/**
+	 * The final score of the player whose Mark is B, i.e. the first player. This is
+	 * a double, in line with the Protocol.
+	 */
+	double blackScore;
+
+	/**
+	 * The final score of the player whose Mark is W, i.e. the second player. This
+	 * is a double, in line with the Protocol, and when counting the scores, will
+	 * get an initial value (Komi) of 0.5.
+	 */
+	double whiteScore;
+
+	/**
+	 * Keeps track of all previous board situations (used for Ko-rule).
+	 */
+	private Set<String> boardSituations;
+
+	/**
+	 * Creates an empty board, gives each intersection its initial number of
+	 * liberties and its list of neighbours, starts the GUI. If gui is set to false,
+	 * the GUI will not be started (useful for testing). Finally initialises the
+	 * board situations as an empty set.
+	 */
+	public Board(int boardSize, boolean gui) {
+		this.firstStone = false;
+		this.boardSize = boardSize;
+		intersecs = new Intersec[boardSize * boardSize];
+		int i = 0;
+		for (int row = 0; row < boardSize; row++) {
+			for (int col = 0; col < boardSize; col++) {
+				intersecs[i] = new Intersec(col, row);
```

Comment:
I'm not completely sure how this works. in line 63 you set `intersecs` to be a `Intersec`-object and than here you set every intersection to be a `Intersec` object as well? I would thought the `intersecs` to be of type Hash?